### ISSUE TITLE
Odin doesn't generate a pdb file or executable with -debug

### ISSUE THREAD

**Issue by creikey** [odin-lang/Odin#2570]
## Context

Windows 11
        Odin: dev-2023-06:788f3c22
        OS:   Windows 11 Home Basic (version: 21H2), build 22000.1936
        CPU:  Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
        RAM:  16276 MiB

## Expected Behavior

A pdb file generates with `odin build . -debug`

## Current Behavior

Nothing is generated, and nothing is printed

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1.  Clone https://github.com/creikey/odin-2d-pbd-physics and checkout to `5aec17bda4676f62b9f3bb41e613d0e4ea96518a` , the commit with the issue at the time of writing this
2. Run `odin build . -debug`
On my setup no pdb of exe file is generated and it silently fails. Somehow this is related to code in the repo, as a few commits ago (I just checked) this doesn't happen.

**Comment by Lperlind** [comment]
I'm guessing the compiler is segfaulting or something then?

**Comment by creikey** [comment]
> I'm guessing the compiler is segfaulting or something then?

I went to go build an odin debug build to test if this was the case, so I did and then tried compiling it, and all the sudden it's generating PDBs correctly. I thought "darn maybe that means it's a release only bug? tough" so I rebuilt the compiler in release and tried building again, all of the sudden everything's peachy and PDBs are generated fine, so I can no longer reproduce this issue. I have no idea what changed or what fixed it, but I'm going to close this now
